src/main/java/seedu/foodrem/logic/commands/tagcommands/ListTagCommand.java

```diff
@@ -0,0 +1,40 @@
+package seedu.foodrem.logic.commands.tagcommands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.foodrem.model.Model.PREDICATE_SHOW_ALL_TAGS;
+
+import java.util.List;
+
+import seedu.foodrem.logic.commands.Command;
+import seedu.foodrem.logic.commands.CommandResult;
+import seedu.foodrem.model.Model;
+import seedu.foodrem.model.tag.Tag;
+
+/**
+ * Lists all the tags available
+ */
+public class ListTagCommand extends Command {
+    public static final String COMMAND_WORD = "listtag";
+
+    public static final String MESSAGE_SUCCESS = "Listed all tags:\n";
+    private String tagList = "";
+
+
+    @Override
+    public CommandResult execute(Model model) {
+        requireNonNull(model);
+        model.updateFilteredTagList(PREDICATE_SHOW_ALL_TAGS);
+        List<Tag> allTags = model.getFilteredTagList();
+
+        StringBuilder allTagsList = new StringBuilder();
+
+        for (Tag tag: allTags) {
+            allTagsList.append(tag.tagName.fullName);
```

Comment:
Instead of using `tag.tagName.fullName`, can we use `tag.getName()`? This is to avoid breaking abstraction barriers and principle of information hiding. 

As a follow up, can we also change the fields `tagName` in `tag` and `fullName` in `TagName` to private fields?

Reply:
Yes will do!